drivers/generic/generic.go

```diff
@@ -127,13 +122,14 @@ func (d *Driver) Create() error {
 			" future will require the ssh agent to contain the appropriate key.")
 	} else {
 		log.Info("Importing SSH key...")
-		// TODO: validate the key is a valid key
-		if err := mcnutils.CopyFile(d.SSHKey, d.GetSSHKeyPath()); err != nil {
-			return fmt.Errorf("unable to copy ssh key: %s", err)
+
+		d.SSHKeyPath = d.ResolveStorePath("id_rsa")
```

Comment:
Good point. Just to clarify, this is the key we're saving so it'd still work for other algos. In that case the hardcoded `id_rsa` filename might be confusing indeed, I've changed it to be the basename of `SSHKey`, hope that's fine.